frontend/src/v5/ui/controls/assigneesSelect/assigneesSelectMenu/assigneesSelectMenu.component.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { formatMessage } from '@/v5/services/intl';
+import { SearchContext, SearchContextType } from '@controls/search/searchContext';
+import { NoResults, SearchInputContainer } from '@controls/searchSelect/searchSelect.styles';
+import { ListSubheader } from '@mui/material';
+import { get, partition } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import { SelectProps } from '@controls/inputs/select/select.component';
+import { useCallback, useContext, useState, useEffect } from 'react';
+import { IUser } from '@/v5/store/users/users.redux';
+import { IJob } from '@/v5/store/jobs/jobs.types';
+import { AssigneesSelectMenuItem } from './assigneesSelectMenuItem/assigneesSelectMenuItem.component';
+import { HiddenSelect, HorizontalRule, SearchInput } from './assigneesSelectMenu.styles';
+
+const isUser = (assignee) => get(assignee, 'user');
+
+const preventPropagation = (e) => {
+	if (e.key !== 'Escape') {
+		e.stopPropagation();
+	}
+};
+
+export const AssigneesSelectMenu = ({
+	open,
+	value,
+	onClick,
+	...props
+}: SelectProps) => {
+	const [users, setUsers] = useState([]);
+	const [jobs, setJobs] = useState([]);
+	const { filteredItems } = useContext<SearchContextType<IJob | IUser>>(SearchContext);
+	const onClickList = useCallback((e) => {
+		preventPropagation(e);
+		onClick?.(e);
+	}, []);
+	useEffect(() => {
+		const [filteredUsers, filteredJobs] = partition(filteredItems, isUser);
+		setUsers(filteredUsers);
+		setJobs(filteredJobs);
+	}, [filteredItems.length]);
+
+	return (
+		// @ts-ignore
+		<HiddenSelect
+			open={open}
+			value={value || []}
+			onClick={onClickList}
+			{...props}
+		>
+			<SearchInputContainer>
+				<SearchInput
+					placeholder={formatMessage({ id: 'searchSelect.searchInput.placeholder', defaultMessage: 'Search...' })}
+					onClick={preventPropagation}
+					onKeyDown={preventPropagation}
+				/>
+			</SearchInputContainer>
+			<ListSubheader>
+				<FormattedMessage id="assigneesSelectMenu.jobsHeading" defaultMessage="Jobs" />
+			</ListSubheader>
+			{jobs.length > 0 && jobs.map(({ _id }) => (
+				<AssigneesSelectMenuItem
+					key={_id}
+					assignee={_id}
+					value={_id}
```

Comment:
Because of the way MUI Selects work the value prop doesn't actually reach the custom component. The select also uses the value prop of its immediate children, which in this case is <AssigneeSelectMenuItem /> and not the <ListItemContainer /> inside AssigneeSelectMenuItem, so you can't set the key or value in there.

From some cursory googling it seems possible to pass value to custom components by using it like a function. Something like this...
![image](https://github.com/3drepo/3drepo.io/assets/62481250/99699a55-bd2f-439c-9304-aa5f32d4dd9d)

But personally I find that more confusing than just passing in a duplicate prop. I don't know what you think or if you can think of a better work around

Reply:
Ahh ok, yeah. I bumped into this issue before, you are right. value gets mapped to `data-value`.
You could potentially access it doing props['data-value'], but I am not sure it's a good idea.
I will mark this as resolved